lib/events/dynamoevents/dynamoevents.go

```diff
@@ -1235,6 +1235,39 @@ func (l *Log) convertFieldsToDynamoMapFormat(ctx context.Context) error {
 	return trace.Wrap(l.migrateMatchingEvents(ctx, filterExpr, transformEvent))
 }
 
+func minInt64(x, y int64) int64 {
+	if x > y {
+		return y
+	}
+	return x
+}
+
+func maxInt64(x, y int64) int64 {
+	if x > y {
+		return x
+	}
+	return y
+}
+
+func (l *Log) approximateOptimalMigrationWorkers() (int64, error) {
+	req := dynamodb.DescribeTableInput{TableName: aws.String(l.Tablename)}
+	table, err := l.svc.DescribeTable(&req)
+	if err != nil {
+		return 0, trace.Wrap(err)
+	}
+
+	// calculate the throughput, accounting for r/w bottlenecks
+	provisioned := table.Table.ProvisionedThroughput
+	if provisioned == nil || provisioned.ReadCapacityUnits == nil || provisioned.WriteCapacityUnits == nil {
+		return maxMigrationWorkers, nil
+	}
```

Comment:
Even if only one of `ReadCapacityUnits` or `WriteCapacityUnits` is set, shouldn't we take that as throughput, instead of falling back to `maxMigrationWorkers`?

Reply:
I would assume this would never happen and the SDK doesn't say it's possible so this is more of a all-hell-breaks-loose sanity check. Doing something more elaborate like that seems like writing unneeded code to optimally handle cases that are a sanity check if the SDK does something wrong.